doc_source/cni-custom-network.md

```diff
@@ -10,6 +10,22 @@ The use cases discussed in this topic require [Amazon VPC CNI plugin for Kuberne
 
 Enabling this feature effectively removes an available elastic network interface \(and all of its available IP addresses for pods\) from each worker node that uses it\. The primary network interface for the worker node is not used for pod placement when this feature is enabled\. You should choose larger instance types with more available elastic network interfaces if you choose to enable this feature\.
 
+Once you enable custom network and apply ENIConfig in different subnet or VPC CIDR, please note that the primary elastic network interface's cannot be used to provide Pod IP addresses. If you are using [Amazon EKS-Optimized Linux AMI](https://docs.aws.amazon.com/en_us/eks/latest/userguide/eks-optimized-ami.html), it is necessary to reduce the max number of Pods that can be scheduled on Node as defined in the [eni-max-pods.txt](https://github.com/awslabs/amazon-eks-ami/blob/master/files/eni-max-pods.txt) config file and restart your worker node manually. This change will let kubelet running on worker node apply new `--max-pods` configuration. For more information, see [kubelet reference](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) in the *Kuberentes documentation*.
```

Comment:
s/elastic network interface's cannot be/elastic network interface cannot be/

Also, I don't believe you need to restart the EC2 worker node in order to get the max pods setting to be picked up. Rather, you should just need to restart kubelet. The EKS worker node AMI contains a bootstrap.sh script that runs on boot of the EC2 worker node. The USE_MAX_PODS env var should be set to "false" and then set KUBELET_EXTRA_ARGS="--max-pods=XXX" to override the value for the instance type from the eni-max-pods.txt file.

Reply:
@jaypipes, thanks for having the advice! I will revise them and update my commit in next couple days.

Reply:
@0xlen Hi! Any chance you'll be able to get to this soon?

Reply:
jaypipes@, thanks for tracking the progress :D and apologize there has a little bit delay. I was getting a try and add more comment in my last commit as you mentioned in the feedback. 

Here is what I tried after adding parameter in CloudFormation provided argument and launched a m5.large instances with AMI `amazon-eks-node-1.12-v20191213 (ami-009ec8dd35b7e044b)` in `ap-southeast-1`:

```
--use-max-pods false --kubelet-extra-args '--max-pods=20'
```

Bootstrap argument for kubelet:
```bash
[ec2-user@ip-192-168-88-248 ~]$ ps auwx | grep max
root      3853  1.2  1.1 1265916 93112 ?       Ssl  14:13   0:04 /usr/bin/kubelet --cloud-provider aws --config /etc/kubernetes/kubelet/kubelet-config.json --allow-privileged=true --kubeconfig /var/lib/kubelet/kubeconfig --container-runtime docker --network-plugin cni --node-ip=192.168.88.248 --pod-infra-container-image=602401143452.dkr.ecr.ap-southeast-1.amazonaws.com/eks/pause-amd64:3.1 --max-pods=20
```

Kubectl:
```bash
$  kubectl describe nodes/ip-192-168-88-248.ap-southeast-1.compute.internal | less
Allocatable:
...
 pods:                        20
```

Reply:
No need to apologize, @0xlen! :)

That looks exactly correct to me.